src/pynwb/tests/test_example_usage_all.py

```diff
@@ -0,0 +1,242 @@
+    nwbfile = NWBFile(
+        session_description="A description of my session",
+        identifier=str(uuid.uuid4()),
+        session_start_time=datetime.datetime.now(datetime.timezone.utc),
+    )
+
+    contacts_table = ContactsTable(
+        description="Test contacts table",
+    )
+    # for demonstration, mix and match different shapes. np.nan means the radius/width/height does not apply
+    contacts_table.add_row(
+        relative_position_in_mm=[10.0, 10.0],
+        contact_id="C1",
+        device_channel=1,
+        shank_id="shank0",
+        plane_axes=[[1.0, 0.0], [0.0, 1.0]],
+        shape="circle",
+        radius_in_um=10.0,
+        width_in_um=np.nan,
+        height_in_um=np.nan,
+    )
+    contacts_table.add_row(
+        relative_position_in_mm=[20.0, 10.0],
+        contact_id="C2",
+        device_channel=2,
+        shank_id="shank0",
+        plane_axes=[[1 / np.sqrt(2), 1 / np.sqrt(2)], [-1 / np.sqrt(2), 1 / np.sqrt(2)]],
+        shape="square",
+        radius_in_um=np.nan,
+        width_in_um=10.0,
+        height_in_um=10.0,
+    )
+
+    # add the object into nwbfile.acquisition for testing
+    # TODO after integration, put this into /general/extracellular_ephys
+    nwbfile.add_acquisition(contacts_table)
+
+    pm = ProbeModel(
+        name="Neuropixels 1.0",
+        description="A neuropixels probe",
+        model="Neuropixels 1.0",
+        manufacturer="IMEC",
+        planar_contour_in_um=[[-10.0, -10.0], [10.0, -10.0], [10.0, 10.0], [-10.0, 10.0]],
+        contacts_table=contacts_table,
+    )
+    # TODO put this into /general/device_models
+    nwbfile.add_device(pm)
+
+    probe = Probe(
+        name="Neuropixels Probe 1",
+        identifier="28948291",
+        probe_model=pm,
+    )
+    nwbfile.add_device(probe)
+
+    pi = ProbeInsertion(
+        position_reference="(AP, ML, DV) = (0, 0, 0) corresponds to bregma at the cortical surface.",
+        hemisphere="left",
+        depth_in_mm=10.0,
+        insertion_position_ap_in_mm=2.0,
+        insertion_position_ml_in_mm=-4.0,
+        insertion_angle_roll_in_deg=-10.0,
+        insertion_angle_pitch_in_deg=0.0,
+        insertion_angle_yaw_in_deg=0.0,
+    )
+
+    channels_table = ChannelsTable(
+        name="Neuropixels1ChannelsTable",  # test custom name
+        description="Test channels table",
+        reference_mode="Referenced to channel 2.",
+        position_reference="(AP, ML, DV) = (0, 0, 0) corresponds to bregma at the cortical surface.",
+        position_confirmation_method="Histology",
+        probe=probe,
+        probe_insertion=pi,
+        target_tables={
+            "contact": probe.probe_model.contacts_table,
+            "reference_contact": probe.probe_model.contacts_table,
+        },
+        # TODO should not need to specify the above
+    )
+
+    # all of the keyword arguments in add_row are optional
+    channels_table.add_row(
+        contact=0,
+        reference_contact=2,
```

Comment:
This is the question about the spec and how it works I think. 
I am confused about how we are adding integers here. The spec says that `contact` inherits from `DynamicTableRegion` and I was expecting that I would have to pass a `DyanmicTableRegion` object here. 

How does it work? Is this a synatactic suggar that creates a region with one entry under the hood? Would it work if I pass a Dynamic Region? I can test the later and I will do now but I wanted to ask in case I got stuck : P

Reply:
Yeah, I tried:

```python
 reference_contact = DynamicTableRegion(
        name="contacts",
        data=[2],
        description="The reference contact",
        table=contacts_table,
    )

```

And then using this as a entry in the row and it does not work.

This is probably a confusion of mine with respect to the schema but I wanted to take this opportunity to learn.

Reply:
`contact` is a column that is a `DynamicTableRegion`. `ContactsTable.add_row(column=1)` adds a single entry `1` to that column (the "data" of the `DynamicTableRegion`). `add_row` takes the indices of the table. You cannot pass in a `DynamicTableRegion`. 

But you could initialize the `ContactsTable` with the `contacts` column with all the row values if you know them in advance:
```python
table = ContactsTable(columns=[
    DynamicTableRegion(
        name="contacts",
        data=[1, 2, 3, 4],
        description="The reference contacts",
        table=contacts_table,
    ), 
    # other columns
)
```

Reply:
Ah, great, so I can think on the row entry as the corresponding entry on a `DynamicTableRegion.data` . Thanks for educating me.

Relevant to this is the following. I remember vaguely that we discussed bipolar electrodes, for that case we would model it with `reference contact`. Is that correct? If that is the case, do you think it make sense to add some comments to the doc of `reference_contact` in that direction?

Reply:
Good idea. I've updated the PR.

Reply:
Thanks.